lib/db/DB.ts

```diff
@@ -58,8 +58,22 @@ class DB {
       const { dialectName } = this.sequelize.connectionManager;
       this.logger.info(`connected to database. host:${host} port:${port} database:${database} dialect:${dialectName}`);
     } catch (err) {
-      this.logger.error('unable to connect to the database', err);
-      throw err;
+      if (err instanceof Sequelize.InvalidConnectionError) {
+        try {
+          const initSequelize = new Sequelize('', this.sequelize.options.username, this.sequelize.options.password, this.sequelize.options);
+          await initSequelize.authenticate();
+          await initSequelize.query(`CREATE DATABASE IF NOT EXISTS ${this.sequelize.options.database};`);
+          await initSequelize.query(`GRANT ALL PRIVILEGES ON ${this.sequelize.options.database}.* TO '${this.sequelize.options.username}'@'%'`);
```

Comment:
Is that necessary?

Reply:
I think I proposed this line earlier because it created the database without the proper permissions for me. I think it was because my `xunion` user didn't have the `GRANT` privilege, only `CREATE`. Without that, even this line doesn't work. I just tried this again commenting out this line and the database was created just fine, so I think we can indeed drop this `GRANT ALL PRIVILEGES` line.